test/compiler/effects.jl

```diff
@@ -346,7 +346,7 @@ function A1_inbounds()
     end
     return r
 end
-@test !Core.Compiler.is_consistent(Base.infer_effects(A1_inbounds))
+@test !Core.Compiler.is_noub(Base.infer_effects(A1_inbounds))
```

Comment:
No, this should taint consistent. :boundscheck is still inconsistent, but `getfield` with boundscheck off (or unknown) taints ub.

Reply:
Well, could you tell me why we even need to taint `:consistent` and/or `:noub` at all? As far as I understand, it would be fine if only `:noinbounds` is tainted, as the `@inbounds` does not have any effect on the `@boundscheck` (, which syntactically appears within the `@inbounds` block).

Reply:
It still has an effect of check-bounds=yes is used, so in the current design :bounds check always needs to taint consistent. Of course with the path dependency I'm adding in the other PR, we may recover consistency for the whole function, but locally, it's inconsistent

Reply:
> It still has an effect of check-bounds=yes is used, so in the current design :bounds check always needs to taint consistent.

Could you please elaborate this please? When `--check-bounds=[yes|no]` is specified, aren't `:boundscheck` regarded as consistent (and thus can be inferred to return `Const([true|false])`)? I guess your concern is that sysimage is not separated by the flag?

Reply:
Yes, that's right. In the current design there's no guarantee that the inference check-bounds matches the runtime check-bounds, so we can't use it for IPO. We may change things around there in the near future, but for the time being that needs to be the semantics.

Reply:
Shouldn't we taint `:consistent` whenever we see `:boundscheck` except the special cased `getfield` call then?